tests/unit/dialog/dialog_core.js

```diff
@@ -7,6 +7,37 @@
 // TODO add teardown callback to remove dialogs
 module("dialog: core");
 
+test( "markup structure", function() {
+	expect( 13 );
+	var element = $( "<div></div>" ).dialog({
+			buttons: [ {
+				text: "Ok",
+				click: $.noop
+			} ]
+		}),
+		widget = element.dialog( "widget" ),
+		titlebar = widget.find( ".ui-dialog-titlebar" ),
+		close = titlebar.find( ".ui-button" ),
+		buttonpane = widget.find( ".ui-dialog-buttonpane" );
+
+	ok( widget.hasClass( "ui-dialog" ), "Widget has class ui-dialog" );
+	ok( widget.hasClass( "ui-widget" ), "Widget has class ui-widget" );
+	ok( widget.hasClass( "ui-widget-content" ), "Widget has class ui-widget-content" );
+	ok( widget.hasClass( "ui-dialog-buttons" ), "Widget has class ui-dialog-buttons" );
+	ok( titlebar.hasClass( "ui-dialog-titlebar" ), "Titlebar has class ui-dialog-titlebar" );
+	ok( titlebar.hasClass( "ui-widget-header" ), "Titlebar has class ui-widget-header" );
+	ok( titlebar.find( ".ui-dialog-title" ).length, "Title hasClass ui-dialog-title" );
```

Comment:
"has class" instead of "hasClass" in the message argument. For the record, I still don't like this approach. The message duplicates what the assertion does, without adding any new information. And that's only necessary since the boolean doesn't provide any context in the case of a failure.

How about a custom assertion that only looks for a given class, but provides useful feedback when it fails?

```js
assert.hasClass( element, "ui-dialog" );
```

With the implementation looking something like this (didn't test):

```js
QUnit.assert.hasClass = function( element, has ) {
  var result = element.hasClass( has );
  QUnit.push( result, has, result ? has : element.attr( "class" ) );
};